compiler/executable/match_/planner/mod.rs

```diff
@@ -42,6 +43,7 @@ pub fn compile(
     variable_registry: &VariableRegistry,
     expressions: &HashMap<Variable, ExecutableExpression<Variable>>,
     statistics: &Statistics,
+    per_call_costs: &impl FunctionCallCostProvider,
```

Comment:
I've told you this before several times - your variable name should _normally_ match your struct/trait name unless you have a good reason not to... `call_cost_provider` is probably fine? If you want to comment that it's "per-call", then you can do that in the trait docstring.

I'd like you to create a list of practices, and write this one down there (along with anything else you remember) and use it as a private checklist for self-review. Maybe that will help you internalise?

Reply:
Was using a hashset here when I wrote it. Will update.